gcc/d/d-decls.cc

```diff
@@ -648,7 +648,8 @@ ClassReferenceExp::toSymbol()
       value->sym = new Symbol();
 
       // Build reference symbol.
-      tree decl = build_artificial_decl(unknown_type_node, NULL_TREE, "C");
+      tree ctype = build_ctype(type);
+      tree decl = build_artificial_decl(TREE_TYPE (ctype), NULL_TREE, "C");
```

Comment:
Ah, wait. I didn't realise that `value->stype` was available (I only looked at `value->type`, which is a null pointer).